### PR TITLE
refs #2926 implemented support for the FtpClient class to restrict co…

### PR DESCRIPTION
**Pull request by davidnich**
…nnections to a certain address family

### CODE REVIEW

File: lib/QoreFtpClient.cpp [hunk 1]
```diff
@@ -553,162 +576,162 @@ struct qore_ftp_private {
...
-      int data_port = atoi(s);
-      if (data.connectINET(host, data_port, timeout_ms, xsink)) {
-         if (!*xsink)
-            xsink->raiseErrnoException("FTP-CONNECT-ERROR", errno, "could not connect to extended passive data port (%s:%d)", host, data_port);
-         return -1;
-      }
-      printd(FTPDEBUG, "EPSV connected to %s:%d\n", host, data_port);
+        // ex: 229 Entering Extended Passive Mode (|||63519|)
+        // get port for data connection
+        printd(FTPDEBUG, "EPSV: %s\n", resp.getBuffer());
```
Comment (sejvlond) [inline lib/QoreFtpClient.cpp:596]: is it intended to have this uncommented?

File: lib/QoreFtpClient.cpp [hunk 2]
```diff
@@ -780,6 +803,17 @@ struct qore_ftp_private {
    DLLLOCAL void clearStats() {
       AutoLocker al(m);
    }
+
+private:
+    DLLLOCAL void init() {
```
Comment (sejvlond) [inline lib/QoreFtpClient.cpp]: init func is unused now, right?

File: lib/QoreFtpClient.cpp [hunk 3]
```diff
@@ -91,247 +91,270 @@ class FtpResp {
...
+    }
+
+    DLLLOCAL QoreHashNode* getControlSocketInfo(ExceptionSink* xsink, bool host_lookup) const {
+        AutoLocker al(m);
+        return control.getSocketInfo(xsink, host_lookup);
+    }
+
+    DLLLOCAL QoreHashNode* getDataSocketInfo(ExceptionSink* xsink, bool host_lookup) const {
+        AutoLocker al(m);
+        return control.getSocketInfo(xsink, host_lookup);
```
Comment (sejvlond) [inline lib/QoreFtpClient.cpp]: shouldn't getDataSocketInfo actually call `data.getSocketInfo` instead of `control.get...`

**Review by sejvlond — CHANGES_REQUESTED** [review]

File: lib/QoreFtpClient.cpp (hunk 1 above)
Comment (davidnich) [reply lib/QoreFtpClient.cpp:596]: yes, `printd()` is changed to nothing in the preprocessor with non-debug builds

**Comment by davidnich** [comment]
@sejvlond all issues resolved - thanks for the review!

**Review by sejvlond — APPROVED** [review]

**Pull request merged**